enhancements/update/clusterversion-osus-signatures.md

```diff
@@ -0,0 +1,148 @@
+---
+title: ClusterVersion-has-option-to-reach-local-OSUS-instance-for-signature-verification
+authors:
+  - "@PratikMahajan"
+reviewers:
+  - "@LalatenduMohanty"
+  - "@wking"
+approvers:
+  - "@LalatenduMohanty"
+api-approvers:
+  - "@deads2k"
+creation-date: 2023-09-27
+last-updated: 2023-10-04
+tracking-link:
+  - https://issues.redhat.com/browse/OTA-916
+---
+
+# ClusterVersion has option to reach local OSUS instance for signature verification
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] Operational readiness criteria is defined
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+CVO currently fetches the upstream urls for signature verification through [cluster-update-keys]. 
+To improve disconnected cluster user experience we need to provide a way for the CVO  to get the required signatures 
+from the OSUS running in the disconnected environment.
+
+
+## Motivation
+
+Today, in the disconnected environments, a user has to clone the signatures config map as well as the release images to update the clusters.
+With [serve image signatures for disconnected environments], OSUS will be able to serve the cloned image signatures locally
+
+Currently, there is no way for the CVO to fetch the signatures through custom URIs. This enhancement defines an API change in the CVO
+to accept a custom update-service URI which can be used to verify release signatures. 
+
+
+### User Stories
+
+* As a cluster administrator, I want the cluster's CVO to add a new signature store URI.
+* As a cluster administrator, I want the cluster's CVO to replace the existing signature store URIs with a user provided URI
+
+### Goals
+
+* To verify the release image signature in disconnected environments using OSUS signatures URI.
+* To replace existing signature stores with custom user-provided stores.
+
+### Non-Goals
+
+* This enhancement does not cover how OSUS running in the disconnected environment gets the signatures and if the 
+signatures are validated in the process
+
+## Proposal
+
+### Workflow Description
+
+1. Cluster admin will have to add the verification URI to CVO
+2. CVO will reach the newly added URI to verify the release images.
+
+### API Extensions
+
+* CVO will grow a new field, an array, which will be populated with the image verification URI.
+* The api extension will replace the existing stores. 
```

Comment:
Not clear to me yet whether array enteries are intended to be accessed in parallel (good for speed when a signature store is down) or serially (good if you'd rather not leak "I care about `$THIS` signature" to a particular mirror in most cases, but are ok leaking that information if higher-priority stores lack the signature you need).  Personally, I'm in favor of parallel, because I can't see "I'd rather not leak, but I'm ok leaking if necessary" being all that popular, but :shrug:, worth laying it out in case other folks have different trade-off considerations and weighting guesses.

Reply:
s/existing stores./existing signature stores in CVO./